virt-v2v/scripts/rhel/run/network_config_util.sh

```diff
@@ -104,9 +123,61 @@ udev_from_nm() {
         fi
 
         echo "SUBSYSTEM==\"net\",ACTION==\"add\",ATTR{address}==\"$(remove_quotes "$S_HW")\",NAME=\"$(remove_quotes "$DEVICE")\""
-    done < "$V2V_MAP_FILE"
+    done
+}
+
+# Create udev rules based on the macToIP mapping + output from parse_netplan_file
+udev_from_netplan() {
+    # Check if netplan command exist
+    if ! command -v netplan >/dev/null 2>&1; then
+        log "Warning: netplan is not installed."
+        return 0
+    fi
+
+    # netplan with root dir
+    netplan_get() {
+        netplan get --root-dir "$NETPLAN_DIR" "$@" 2>/dev/null
+    }
+
+    # Loop over all interface names and treturn the one with target_ip, or null
+    find_interface_by_ip() {
+        target_ip="$1"
+
+        # Loop through all interfaces and check for the given IP address
+        netplan_get ethernets | grep -Eo "^[^[:space:]]+[^:]" | while read IFNAME; do
+            if netplan_get ethernets."$IFNAME".addresses | grep -q "$target_ip"; then
+                echo "$IFNAME"
```

Comment:
either we should `return` here
or later we need to check that we only found one ifname

Reply:
added return here